src/openloco/things/CreateVehicle.cpp

```diff
@@ -0,0 +1,966 @@
+#include "../company.h"
+#include "../companymgr.h"
+#include "../core/Optional.hpp"
+#include "../date.h"
+#include "../game_commands.h"
+#include "../management/Expenditures.h"
+#include "../map/tile.h"
+#include "../objects/objectmgr.h"
+#include "../objects/road_object.h"
+#include "../objects/track_object.h"
+#include "../objects/vehicle_object.h"
+#include "../ui/WindowManager.h"
+#include "thingmgr.h"
+#include "vehicle.h"
+#include <numeric>
+#include <utility>
+
+using namespace openloco;
+using namespace openloco::interop;
+using namespace openloco::objectmgr;
+
+namespace openloco::things::vehicle
+{
+    constexpr uint32_t max_orders = 256000;
+    constexpr auto max_ai_vehicles = 500;
+    constexpr auto max_num_car_components_in_car = 4;           // TODO: Move to vehicle_object
+    constexpr auto num_vehicle_components_in_car_component = 3; // Bogie bogie body
+    constexpr auto num_vehicle_components_in_base = 4;          // head unk_1 unk_2 tail
+    constexpr auto max_num_vehicle_components_in_car = num_vehicle_components_in_car_component * max_num_car_components_in_car;
+    constexpr thing_id_t allocated_but_free_routing_station = -2; // Indicates that this array is allocated to a vehicle but no station has been set.
+
+    static loco_global<company_id_t, 0x009C68EB> _updating_company_id;
+    static loco_global<uint16_t, 0x009C68E0> gameCommandMapX;
+    static loco_global<uint16_t, 0x009C68E2> gameCommandMapY;
+    static loco_global<uint16_t, 0x009C68E4> gameCommandMapZ;
+    static loco_global<string_id, 0x009C68E6> gGameCommandErrorText;
+    static loco_global<uint8_t, 0x009C68EA> gGameCommandExpenditureType; // premultiplied by 4
+    static loco_global<uint8_t, 0x009C68EE> _errorCompanyId;
+    static loco_global<map::tile_element*, 0x009C68D0> _9C68D0;
+    static loco_global<ColourScheme, 0x01136140> _1136140; // primary colour
+    static loco_global<int32_t, 0x011360FC> _11360FC;
+    static loco_global<openloco::vehicle_head*, 0x01136240> _backupVeh0;
+    static loco_global<int16_t, 0x01136248> _backup2E;
```

Comment:
Can these backups be locals?

Reply:
Leave until game command 0 and 6 are implemented.